### PR TITLE
docs(examples): make Stitches example render properly the SSR pass

### PR DESCRIPTION
**Pull request by matthova**
This PR patches the the [stitches](https://github.com/remix-run/remix/tree/main/examples/stitches) example to properly set server-rendered styles. Previously, the styles (other than the theme) were not set on initial render. I'm not the biggest fan of the regex I wrote, please advise if a better replacement comes to mind.

### CODE REVIEW

File: examples/stitches/app/entry.server.tsx [hunk 1]
```diff
@@ -11,15 +11,16 @@ export default function handleRequest(
...
+  let sheet = getCssText();
 
   let markup = renderToString(
     <ServerStyleContext.Provider value={sheet}>
       <RemixServer context={remixContext} url={request.url} />
     </ServerStyleContext.Provider>
   );
 
-  markup = markup.replace("__STYLES__", sheet);
+  sheet = getCssText();
```
Comment (LexSwed) [inline examples/stitches/app/entry.server.tsx:22]: The change of `const sheet = getCssText();` is unnecessary, repeating twice `getCssText()`

Comment (matthova) [reply examples/stitches/app/entry.server.tsx:22]: Or 
```ts
  const sheet = getCssText();

  let markup = renderToString(
    <ServerStyleContext.Provider value={sheet}>
      <RemixServer context={remixContext} url={request.url} />
    </ServerStyleContext.Provider>
  );

  const updatedSheet = getCssText();
  markup = markup.replace(/<style id=\"stitches\">.*\<\/style>/g, `<style id="stitches">${updatedSheet}</style>`);
```

Comment (LexSwed) [reply examples/stitches/app/entry.server.tsx:22]: The later looks good, it was not immediately clear to me why calling it twice 👀 And still not clear though 😅  If `getCssText` is only returning a string, why passing it to the initial markup render? It's not gonna be used during first render anyway, right?
I mean changing this to:
```js
  let markup = renderToString(
    <RemixServer context={remixContext} url={request.url} />
  );

  const cssText = getCssText();
  markup = markup.replace(/<style id=\"stitches\">.*\<\/style>/g, `<style id="stitches">${cssText}</style>`);
```
will work for the resulting markup anyway, right?

Comment (matthova) [reply examples/stitches/app/entry.server.tsx]: Good eye. This pattern had been copy/pasted from the `styled-components` example, where it appears the server context isn't necessary at all. I've updated it to be a bit cleaner.

**Review by LexSwed — APPROVED** [review]
Looks sweet to me 🍰

**Review by kentcdodds — APPROVED** [review]
Looks like this is fixing some copy/paste mistake. Thanks!

**Comment by coreybruyere** [comment]
@matthova I'm getting FOUC when following your example? Care to take a look here? https://github.com/coreybruyere/california-custom-tile/pull/2

**Comment by daphnesmit** [comment]
Quick dumb question; where would you typically but the : 
```
  globalStyles();
```

of stitches?

**Pull request merged**